src/main/kotlin/Assigment.kt

```diff
@@ -11,7 +11,7 @@ class Assigment(
             throw IllegalArgumentException("Nome do curso não pode ser vazio")
         if (code.isBlank())
             throw IllegalArgumentException("Código não pode ser vazio")
-        if (!code.all { it.isLowerCase() && it.isLetter() || it == '-' })
-            throw IllegalArgumentException("Código deve ser composto somente por letras minusculas e -")
+        if (!code.matches("^(?:[a-z]-?)+[^-]\$".toRegex()))
```

Comment:
really nice approach!! 

did you get this regex from trello ?  I think that i left there to help everyone.